feat: add entities for user and meal

gusdls opened the pull request:
Add TypeORM entities for user, meal, food and nutrient. And change the id column of nutrient entity from `uuid` to `int`

gusdls committed:
feat(entity): make entities for user and food

gusdls committed:
feat(entity): change columns of nutrient

gusdls committed:
fix(nutrient): change id column to int

lukekeum commented:
그 user.entity.ts에 deletedAt 추가 해줄 수 있낭 ,, 나중에 필요하면 그냥 삭제 말고 soft-delete하려고

lukekeum commented on backend/src/entities/user.entity.ts:
```diff
@@ -0,0 +1,34 @@
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
```
@gusdls 여기서 deletedAt 추가해줘 데코레이터는 DeleteDateColumn일거임

lukekeum replied:
@gusdls 여기서 deletedAt column 추가해줘. 데코레이터는 @DeleteDateColumn일겨

lukekeum commented on backend/src/entities/food.entity.ts:
```diff
@@ -0,0 +1,21 @@
+  photo: string;
+
+  @ManyToMany(() => Nutrient, (nutrient) => nutrient.foods)
+  nutrients: Nutrient[];
```
그 여기에 다대다 관계 말고 일대다 쓰는게 맞을거 같음.

gusdls replied:
여러 영양소가 여러 음식에 할당되면 다대다가 맞지 않아요?

gusdls replied:
여러 음식에 같은 영양소가 있을 수 있고 한 음식이 여러 영양소를 가질 수 있으니 다대다가 맞지 않을까요?

lukekeum commented on backend/src/entities/nutrient.entity.ts:
```diff
@@ -0,0 +1,14 @@
+  name: string;
+
+  @ManyToMany(() => Food, (food) => food.nutrients)
+  foods: Food[];
```
영양소로 음식 조회할 일은 없을거라, 없애는게 맞을수도

lukekeum commented on backend/src/entities/meal.entity.ts:
```diff
@@ -0,0 +1,17 @@
+  id: string;
+
+  @Column()
+  rating: number;
```
평점을 숫자로 해놓으면, 누가 평점 했는지를 모름. (1인 1표가 원칙인데 확인 불가). 그래서 그 댓글 처럼 엔티티 구현 가능할까 ?

rating.entity
- rating
- usrId
- mealId

gusdls replied:
아니면 급식 리뷰를 만드는 것도 괜찮을 거 같지 않아요?

gusdls replied:
차라리 급식 리뷰를 만드는 건 어떨까요?

lukekeum replied:
어엉 그런 느낌으로 가자구

lukekeum commented on backend/src/entities/food.entity.ts:
```diff
@@ -0,0 +1,21 @@
+  nutrients: Nutrient[];
+
+  @ManyToMany(() => Meal, (meal) => meal.foods)
+  meals: Meal[];
```
이거는 나중에 음식 가지고 날짜 조회할 일 있어서 냅두는게 좋을듯 굿굿

gusdls committed:
feat(review): add an entity for review and fix something

add an entity for review
add deletedAt column to user entity
make the relation between food and
nutrient to uni-directional

gusdls commented:
급식 리뷰 엔티티 만들어 놨고 나머지는 다 해결했어요
